substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/heap/PhysicalMemory.java

```diff
@@ -107,6 +112,32 @@ public static UnsignedWord size() {
         return cachedSize;
     }
 
+    /**
+     * Returns the amount of used physical memory in bytes, or -1 if not supported yet.
+     */
+    public static long usedSize() {
+        if (Containers.isContainerized()) {
+            return Containers.memoryUsageInBytes();
+        }
+        if (Platform.includedIn(Platform.LINUX.class)) {
+            try {
+                List<String> lines = Files.readAllLines(Paths.get("/proc/meminfo"));
+                for (String line : lines) {
+                    if (!line.contains("MemAvailable")) {
+                        continue;
+                    }
+                    String memAvailable = line.replaceAll("\\D", "");
+                    if (!memAvailable.isEmpty()) {
+                        return size().rawValue() - Long.parseLong(memAvailable);
+                    }
+                }
+            } catch (IOException e) {
+                return -1;
+            }
+        }
+        return -1;
```

Comment:
I believe a reasonable implementation would be to use the OS bean here for other platforms, rather than keep unimplemented. YMMV.

Reply:
I've adopted your suggestion and am now using `OperatingSystemMXBean` if not on linux and `UseContainerSupport` is set.